### DISCUSSION TITLE
Is it possible to use astro db with github pages deploy?

### DISCUSSION THREAD

**Discussion by isimmons** [withastro/docs#7537] [Ideas]
I've gone through the setup for github pages deploy and astro studio deploy and everything seems to be working except that for the --remote flag in the build command, Astro CLI needs to log in to Studio to handle verifying and pushing db changes.

Is there any other way to authenticate before running the `astro build --remote` command?

**Comment by Fryuni** [comment]
Yes, you can follow [this guide](https://docs.astro.build/en/recipes/studio/) on how to deploy connecting to Studio.

More specifically, to authenticate to Astro Studio you need an [app token](https://docs.astro.build/en/recipes/studio/#create-a-studio-app-token). You then add that app token in an `ASTRO_STUDIO_APP_TOKEN` environment variable when running `astro build --remote`.

For GH pages you need to add that as a secret and then expose the secret as an env var with:
```yaml
env:
  ASTRO_STUDIO_APP_TOKEN: ${{secrets.ASTRO_STUDIO_APP_TOKEN }}
```

**Reply by isimmons** [reply]
Ah well that makes sense. I have a studio.yml workflow and a deploy.yml workflow. I had the env set in the studio.yml but not in the deploy.yml.

Thanks

**Comment by ArmandPhilippot** [comment]
We're closing stale and outdated discussion to keep only the discussions still relevant to the current project version and/or the current docs state.

If you think this is a mistake and that this discussion should remain open, please feel free to ping me, I'll take a second look at it.

Thank your for your contribution!
